src/components/Personality/PersonalityCreateSearch.tsx

```diff
@@ -28,14 +28,16 @@ const PersonalityCreateSearch = ({
             ...personality,
             ...personalityCreated,
         };
-        const createClaim = () => {
+        const createClaim = async () => {
             selectPersonality(newPersonality);
             setIsFormSubmitted(false);
         };
 
         // Redirect to personality list in case _id is not present
         const path = slug ? `/personality/${slug}` : "/personality";
-        selectPersonality !== null ? createClaim() : router.push(path);
+        selectPersonality !== null
```

Comment:
**suggestion:** in this case is better to use an if statement, the ternary operator is hard to read and not suitable if you're not assigning a value to a variable.